include/cbor/stl.h

```diff
@@ -571,6 +572,76 @@ struct traits<std::map<KeyType, ValueType>>
   }
 };
 
+/**
+ * Specialization for std::shared_ptr.
+ */
+template <typename PointedType>
+struct traits<std::shared_ptr<PointedType>>
+{
+  using Type = std::shared_ptr<PointedType>;
+  template <typename Data>
+  static result serializer(const std::shared_ptr<PointedType>& v, Data& data)
+  {
+    if (v == nullptr)
+    {
+      return to_cbor(nullptr, data);
+    }
+    return to_cbor(*v, data);
+  }
+
+  template <typename Data>
+  static result deserializer(Type& v, Data& data)
+  {
+    if (data.isNull())
+    {
+      v = nullptr;
+      return data.advance(1);
+    }
+    else
+    {
+      v = std::make_shared<PointedType>();
+      auto res = from_cbor(*v, data);
+      return res;
+    }
+    return false;
```

Comment:
Why not avoid the `else` and only leave the `if` to handle the special case?:
``` c++
    if (data.isNull())
    {
      v = nullptr;
      return data.advance(1);
    }

    v = std::make_shared<PointedType>();
    return from_cbor(*v, data);
```

Reply:
Much cleaner. I've fixed this.